File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/CatherbyTeleportSpell.java
package plugin.interaction.combat.magic.lunar.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/28/2017
 */
public class CatherbyTeleportSpell implements TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 87;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 3, LAW_RUNE, 3, WATER_RUNE, 10);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2804, 3433, 0);
	}
	
	@Override
	public int spellId() {
		return 44;
	}
	
	@Override
	public double exp() {
		return 93;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/CatherbyTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/28/2017
 */
class CatherbyTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 87
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 3, MagicConstants.LAW_RUNE, 3, MagicConstants.WATER_RUNE, 10)
    }

    override fun destination(): WorldTile {
        return WorldTile(2804, 3433, 0)
    }

    override fun spellId(): Int {
        return 44
    }

    override fun exp(): Double {
        return 93.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}